File: provider/src/main/java/com/raygun/raygun4android/messages/rum/RaygunRUMMessage.java
package com.raygun.raygun4android.messages.rum;

public class RaygunRUMMessage {
    private RaygunRUMDataMessage[] eventData;

    public RaygunRUMDataMessage[] getEventData() {
        return eventData;
    }

    public void setEventData(RaygunRUMDataMessage[] eventData) {
        this.eventData = eventData;
    }
}

File: provider/src/main/java/com/raygun/raygun4android/messages/rum/RaygunRUMMessage.kt
package com.raygun.raygun4android.messages.rum

data class RaygunRUMMessage(
    var eventData: Array<RaygunRUMDataMessage> = emptyArray(),
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as RaygunRUMMessage

        return eventData.contentEquals(other.eventData)
    }

    override fun hashCode(): Int = eventData.contentHashCode()
}
